source/LocationFilter.cpp

```diff
@@ -190,7 +190,7 @@ void LocationFilter::Save(DataWriter &out) const
 			{
 				if(!governmentsBlacklist.empty())
 					for(const Government *government : governmentsBlacklist)
-						out.Write(government->GetName());
+						out.Write("!" + government->GetName());
```

Comment:
I would flip the order: if there is a government on the include list, write that one. If the include list is empty, then write the blacklist.